pkgs/tools/misc/ckb/default.nix

```diff
@@ -1,47 +1,54 @@
-{ stdenv, fetchFromGitHub, substituteAll, udev
-, pkgconfig, qtbase, qmake, zlib, kmod }:
+{ stdenv, fetchFromGitHub, cmake, pkgconfig, systemd, udev, substituteAll
+, qtbase, zlib, kmod, libpulseaudio, libappindicator-gtk2, gnused }:
 
 stdenv.mkDerivation rec {
-  version = "0.2.9";
+  version = "0.3.1";
   name = "ckb-next-${version}";
 
   src = fetchFromGitHub {
     owner = "ckb-next";
     repo = "ckb-next";
     rev = "v${version}";
-    sha256 = "0hl41znyhp3k5l9rcgz0gig36gsg95ivrs1dyngv45q9jkr6fchm";
+    sha256 = "1jivz6k4qkjykps5z80dhbs2pgn6w2nwam52w56yg8393zskknn8";
   };
 
   buildInputs = [
-    udev
-    qtbase
-    zlib
+    systemd udev qtbase zlib gnused
+    libpulseaudio libappindicator-gtk2
   ];
 
   nativeBuildInputs = [
     pkgconfig
-    qmake
+    cmake
   ];
 
   patches = [
-    ./ckb-animations-location.patch
     (substituteAll {
       name = "ckb-modprobe.patch";
       src = ./ckb-modprobe.patch;
       inherit kmod;
     })
+    ./daemon.patch
```

Comment:
Are you able to document these patches and submit them upstream if possible?

Reply:
Are these comments sufficient? The part that calls modprobe could be fixed to just load the module directly in C, and the cmake files could be changed to take some options for installation directories instead of hardcoding, but these are needed as it is now.